### PR TITLE
Modified class ERst, added 1 new test to test_ERst

### PR DESCRIPTION
**Pull request by tskille**
subroutine 'seqnumForSeparateFilename'  in ERst has been updated and this is no longer using `boost::regex`. One new test added which to test_ERst which is testing separate formatted and unformatted restart file (SPE1_TESTCASE.F0025 and SPE1_TESTCASE.X0025).

### CODE REVIEW

File: src/opm/io/eclipse/ERst.cpp [hunk 1]
```diff
@@ -27,27 +27,33 @@
 #include <stdexcept>
 #include <string>
 
-#include <boost/regex.hpp>
+#include <iostream>
```
Comment (bska) [inline src/opm/io/eclipse/ERst.cpp]: I don't think you need `<iostream>` here.

File: src/opm/io/eclipse/ERst.cpp [hunk 2]
```diff
@@ -27,27 +27,33 @@
...
+        if (filename.find(".F") != std::string::npos) {
+            p = filename.find_last_of(".F");
+        } else if (filename.find(".X") != std::string::npos) {
+            p = filename.find_last_of(".X");
+        } else {
+            OPM_THROW(std::invalid_argument, errMessage);
+        }
+        
+        try {
+            return std::stoi(filename.substr(p+1,filename.size() - p-1));
```
Comment (bska) [inline src/opm/io/eclipse/ERst.cpp]: You don't need the second argument here.  `filename.substr(p + 1)` will extract the same substring.

**Comment by bska** [comment]
Having thought about this some more, I've decided to merge this in its current form.  Once we raise the minimum compiler version to GCC 4.9 or later we can reintroduce the method using regular expressions, but based on `std::regex` instead.

**Comment by joakim-hove** [comment]
I have a feeling the defacto minimum compiler is already at 4.9 - or can anyone confirm compilation on 4.8.x

**Comment by alfbr** [comment]
Yes, GCC 4.8 failed a long time ago, so we have had a de-facto minimum requirement of GCC 4.9 for some time already. Even this seems to break as we speak:
https://github.com/OPM/opm-material/issues/344

**Comment by blattms** [comment]
We should have some kind of compiler support matrix in the wiki

| OPM Version | 2019.10 |  ... |
| --- | --- | --- |
| min g++ version | 4..9 | ... |

**Pull request merged**